docs/decisions/0021-json-serializable-custom-types.md

```diff
@@ -0,0 +1,163 @@
+        {
+            return JsonSerializer.Serialize(value);
+        }
+    }
+```
+
+## Considered Options
+
+**1. Use TypeConverter for primitive types and native serialization methods for custom types**
+For this option, an attribute will not be required on the custom type by default as `JsonSerializer.Deserialize<object>()` will be used to convert from a string, and `JsonSerializer.Serialize()` used to convert to a string.
+
+Preserving the use of the native `TypeConverter`s for primitive types is necessary to prevent any lossy conversions.
+
+This could change the `GetTypeConverter()` method to look like the following, where before the `JsonConverterWrapper` wasn't needed and the function returned a `TypeConverter` instead of a `JsonConverter`:
+
+```csharp
+    private static JsonConverter? GetTypeConverter(Type targetType)
+    {
+        if (targetType == typeof(byte)) { return new JsonConverterWrapper(new ByteConverter()); }
+        if (targetType == typeof(sbyte)) { return new JsonConverterWrapper(new SByteConverter()); }
+        if (targetType == typeof(bool)) { return new JsonConverterWrapper(new BooleanConverter()); }
+        if (targetType == typeof(ushort)) { return new JsonConverterWrapper(new UInt16Converter()); }
+        if (targetType == typeof(short)) { return new JsonConverterWrapper(new Int16Converter()); }
+        if (targetType == typeof(char)) { return new JsonConverterWrapper(new CharConverter()); }
+        if (targetType == typeof(uint)) { return new JsonConverterWrapper(new UInt32Converter()); }
+        if (targetType == typeof(int)) { return new JsonConverterWrapper(new Int32Converter()); }
+        if (targetType == typeof(ulong)) { return new JsonConverterWrapper(new UInt64Converter()); }
+        if (targetType == typeof(long)) { return new JsonConverterWrapper(new Int64Converter()); }
+        if (targetType == typeof(float)) { return new JsonConverterWrapper(new SingleConverter()); }
+        if (targetType == typeof(double)) { return new JsonConverterWrapper(new DoubleConverter()); }
+        if (targetType == typeof(decimal)) { return new JsonConverterWrapper(new DecimalConverter()); }
+        if (targetType == typeof(TimeSpan)) { return new JsonConverterWrapper(new TimeSpanConverter()); }
+        if (targetType == typeof(DateTime)) { return new JsonConverterWrapper(new DateTimeConverter()); }
+        if (targetType == typeof(DateTimeOffset)) { return new JsonConverterWrapper(new DateTimeOffsetConverter()); }
+        if (targetType == typeof(Uri)) { return new JsonConverterWrapper(new UriTypeConverter()); }
+        if (targetType == typeof(Guid)) { return new JsonConverterWrapper(new GuidConverter()); }
+
+        if (targetType.GetCustomAttribute<JsonConverterAttribute>() is JsonConverterAttribute tca &&
+            Type.GetType(tca.ConverterType.Name, throwOnError: false) is Type converterType &&
+            Activator.CreateInstance(converterType) is JsonConverter converter)
```

Comment:
Can you explain why this must be a JsonConverter?

Reply:
it felt like the more natural choice since we're now defaulting to serialization, but you're right it doesn't have to be. 
thinking more about it, it will be less disruptive if we leave this alone as `TypeConverter` and instead just default to a `SimpleTypeConverter` like the one shown in the above example.